### PR TITLE
rewrite protect to use message.original

### PR DESCRIPTION
**Pull request by rukai**
closes https://github.com/shotover/shotover-proxy/issues/476

Currently the protect transform serializes the encrypted values to bincode and then stores the result in a MessageValue::Bytes.
The MessageValue::Bytes fails to round trip through cassandra because the current cassandra AST lacks a way to represent binary blobs.
Previously that didnt cause any problems because the query_values message details never got assembled back into the outgoing message.
But as of this PR this functionality actually works so this PR reverts the serialization to use serde instead of bincode as we can easily round trip through a string value.

Once we get a proper cql ast we can revert back to bincode and MessageValue::Bytes.

### CODE REVIEW

**Review by benbromhead — APPROVED** [review]

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 1]
```diff
@@ -68,23 +68,23 @@ fn decrypt(ciphertext: Vec<u8>, nonce: Nonce, sym_key: &Key) -> Result<MessageVa
...
         secretbox::open(&ciphertext, &nonce, sym_key).map_err(|_| anyhow!("couldn't open box"))?;
     //TODO make error handing better here - failure here indicates a authenticity failure
-    // let decrypted_value: Value =
-    //     serde_json::from_slice(decrypted_bytes.as_slice()).map_err(|_| anyhow!("couldn't open box"))?;
-    let decrypted_value: MessageValue =
-        bincode::deserialize(&decrypted_bytes).map_err(|_| anyhow!("couldn't open box"))?;
+    let decrypted_value: MessageValue = serde_json::from_slice(decrypted_bytes.as_slice())
+        .map_err(|_| anyhow!("couldn't open box"))?;
+    // let decrypted_value: MessageValue =
+    //     bincode::deserialize(&decrypted_bytes).map_err(|_| anyhow!("couldn't open box"))?;
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:74]: Is the intention all this commented out code will be replaced when the Cassandra AST is ready?

Comment (rukai) [reply shotover-proxy/src/transforms/protect/mod.rs:74]: yeah

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 2]
```diff
@@ -156,25 +156,77 @@ impl ProtectConfig {
...
     async fn transform<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
+        // encrypt the values included in any INSERT or UPDATE queries
         for message in message_wrapper.messages.iter_mut() {
-            if let MessageDetails::Query(qm) = &mut message.details {
-                // Encrypt the writes
-                if QueryType::Write == qm.query_type && qm.namespace.len() == 2 {
-                    if let Some((_, tables)) =
-                        self.keyspace_table_columns.get_key_value(&qm.namespace[0])
+            if let Some(namespace) = message.namespace() {
+                if namespace.len() == 2 {
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:211]: Won't a namespace always have the keyspace and column? Once we fix the `USE keyspace` handling of course.

**Review by conorbros — APPROVED** [review]

File: shotover-proxy/src/transforms/protect/mod.rs (hunk 2 above)
Comment (rukai) [reply shotover-proxy/src/transforms/protect/mod.rs:211]: I'm just maintaining the existing functionality, I think when we implement proper `USE keyspace` we can go through and make sure the namespace implementation and usage is sensible.

**Pull request merged**